src/common/snippets/src/lowered/pass/insert_tail_loop.cpp

```diff
@@ -13,9 +15,74 @@ namespace snippets {
 namespace lowered {
 namespace pass {
 
+std::shared_ptr<op::LoopEnd> InsertTailLoop::create_tail_loop(LinearIR& linear_ir,
+                                                              LinearIR::constExprIt vector_begin,
+                                                              LinearIR::constExprIt vector_end,
+                                                              LinearIR::constExprIt& tail_begin,
+                                                              LinearIR::constExprIt& tail_end,
+                                                              const std::shared_ptr<op::LoopEnd>& vector_loop_end,
+                                                              bool need_vector_loop,
+                                                              size_t tail_size,
+                                                              const std::vector<int64_t>& tail_finalization_offsets) {
+    // tail is required => transform the body into a tail representation
+    // tail loop is fake loop because for tail we should calculate only
+    // finalization offsets which are supported by LoopEnd.
+    if (need_vector_loop) {
+        auto vector_loop_deep_copy = LinearIR::deep_copy_range(vector_begin, vector_end);
+        auto is_par_or_res = [](const ExpressionPtr& expr) {
+            return is_type<ov::op::v0::Parameter>(expr->get_node()) ||
+                   is_type<ov::op::v0::Result>(expr->get_node());
+        };
+        // Note: It's illegal to insert Parameter or Result to the IR, but they can appear inside vector loop
+        //  So we have to remo them before injecting tail loop into linear_ir
+        auto to_erase = std::remove_if(vector_loop_deep_copy.begin(), vector_loop_deep_copy.end(), is_par_or_res);
```

Comment:
Why do we need to erase them explicitly? `Results` should be outside of the loop, because there is a special pass for that (`lowered::pass::MoveResultOutOfLoop`), and `Parameters` must be outside by construction (they are always the 1st N nodes in LIR), so `LoopBegin` should be simply inserted after them.
Could you please clarify this point?